source/magpie/group.d

```diff
@@ -111,6 +120,101 @@ public:
+    +/
+    string display(T)(T[] groupIndex = [],bool getStr = false, int termianlw = 0)
+    {
+        import std.array: appender;
+        auto retstr = appender!(string);
+
+        int[] pos;
+        if(groupIndex.length == 0)
+        {
+            pos.length = groups.length;
+            foreach(i; 0 .. groups.length)
+                pos[i] = cast(int)i;
+        }
+        else static if(is(T == int))
+        {
+            import std.algorithm: reduce, max, min;
+            assert(groupIndex.reduce!max < groups.length, "Index out of bound");
+            assert(groupIndex.reduce!min > -1, "Index out of bound");
+            pos = groupIndex;   
+        }
+        else static if(is(T == string))
+        {
+            pos.length = 1;
+            int indxpos = getGroupPosition(groupIndex);
+            assert(indxpos > -1, "Group not found");
+            pos[0] = indxpos;
+        }
+        else static if(is(T == string[]))
+        {
+            pos.length = groupIndex.length;
+            foreach(i, ele; groupIndex)
+            {
+                int indxpos = getGroupPosition(ele);
+                assert(indxpos > -1, "Group not found");
+                pos[i] = indxpos;
+            }
+        }
+        else
+        {
+            assert(0, "Group Indexes must be an array of integer, 1D array of string or 2D array of string");
+        }
+
+        DataFrame!(true, GrpRowType) displayHelper;
+        displayHelper.indx.column = grpIndex.column;
+        displayHelper.indx.row.titles = grpIndex.row.titles;
+        displayHelper.indx.row.index = grpIndex.row.index;
+        displayHelper.indx.row.codes.length = grpIndex.row.codes.length;
+
+        foreach(i; 0 .. pos.length)
+        {
+            foreach(j; 0 .. displayHelper.indx.row.codes.length)
+            {
+                displayHelper.indx.row.codes[j] = grpIndex.row.codes[j][elementCountTill[pos[i]] .. elementCountTill[pos[i] + 1]];
+            }
+
+            static foreach(j; 0 .. GrpRowType.length)
+                displayHelper.data[j] = data[j][elementCountTill[pos[i]] .. elementCountTill[pos[i] + 1]];
+
+            displayHelper.rows = elementCountTill[pos[i] + 1] - elementCountTill[pos[i]];
+            string display = displayHelper.display(true, termianlw);
+
+            if(getStr)
+            {
+                retstr.put(display);
+                retstr.put("\n");
+            }
+            else
+            {
+                import std.stdio: writeln;
+                writeln("Group: ", groups[i]);
+                writeln("Group Dimension: [ ", elementCountTill[pos[i] + 1] - elementCountTill[pos[i]]," X ", GrpType.length, " ]");
+                writeln(display);
+            }
+        }
+
+        return retstr.data;
+    }
+
+    /++
+    string[][] getGroups() @property
+    Description: Get the generated string array containing all the groups
+    +/
+    string[][] getGroups() @property
+    {
+        return groups;
```

Comment:
why is `groups` not a public field in the first place?

Reply:
Addressed in 6517b4a 
I'll find more place where this can be done

Reply:
I had a different approach planned for this.

Reply:
OK.